### PR TITLE
feat: add `biome.configPath` option

### LINKED ISSUE #28: Add `configPath` option
This option will allow the extension to resolve the configuration from a different path, as opposed to the default configuration resolution.

This option will pass `--config-path` to the CLI, so it will behave the same.

### PR DESCRIPTION
**Pull request by nhedger**
fixes #28

### CODE REVIEW

**Comment by nhedger** [comment]
I mistakenly thought that the __print_socket command already accepted a `--config-path` argument but this is not the case. Going forward with this PR is going to require modifying the CLI to allow this arg.

**Comment by ematipico** [comment]
I completely forgot that the VSCode extension bootstraps the daemon in a different way. I think we can add it easily to the LSP.

**Comment by nhedger** [comment]
Yep, I'll take care of it.

File: package.json [hunk 1]
```diff
@@ -92,6 +92,11 @@
 					"type": ["boolean", "null"],
 					"default": null,
 					"markdownDescription": "Enable/Disable Biome handling renames in the workspace. (Experimental)"
+				},
+				"biome.configPath": {
+					"type": ["string", "null"],
+					"default": null,
```
Comment (Malix-off) [inline package.json:98]:
```suggestion
					"type": "string",
					"default": "",
```

File: src/main.ts [hunk 2]
```diff
@@ -490,7 +490,21 @@ async function getSocket(
 	outputChannel: OutputChannel,
 	command: string,
 ): Promise<string> {
-	const process = spawn(command, ["__print_socket"], {
+	const extraArgs = [];
+
+	const configPath = workspace
+		.getConfiguration()
+		.get<string | null>("biome.configPath");
```
Comment (Malix-off) [inline src/main.ts:497]:
```suggestion
		.get<string>("biome.configPath");
```

File: src/main.ts [hunk 3]
```diff
@@ -490,7 +490,21 @@ async function getSocket(
...
+	if (configPath && configPath.length > 0) {
```
Comment (Malix-off) [inline src/main.ts:498]:
```suggestion
	if (configPath) {
```

## Rationale

```js
Boolean("") // false
Boolean(" ") // true
```

**Review by Malix-off — CHANGES_REQUESTED** [review]
Comply with #101

**Comment by Malix-off** [comment]
@nhedger can you provide reasons as of why this PR has been closed?

**Comment by nhedger** [comment]
We'll be going a different way that doesn't involve passing the setting as an arg.

**Comment by skipjack** [comment]
@nhedger is there a way to do this now? I don't see that alternative linked here or in the close issue relating to it.

**Comment by skipjack** [comment]
Disregard, I see [your comment here][1] now... trying that out.

[1]: https://github.com/biomejs/biome-vscode/issues/28#issuecomment-2358605741

**Pull request closed without merging**